auth/service.go

```diff
@@ -792,42 +792,28 @@ func (svc service) UnassignUsers(ctx context.Context, token, id string, userIds
 		return err
 	}
 
-	if relation != "" {
-		pr.Permission = SwitchToPermission(relation)
-		if err := svc.Authorize(ctx, pr); err != nil {
-			return err
-		}
-
-		if err := svc.removeDomainPolicies(ctx, id, relation, userIds...); err != nil {
-			return errors.Wrap(errRemovePolicies, err)
-		}
-		return nil
-	}
 	pr.Permission = AdminPermission
 	if err := svc.Authorize(ctx, pr); err != nil {
+		pr.SubjectKind = UsersKind
 		// User is not admin.
-		var ids []string
 		for _, userID := range userIds {
-			if err := svc.Authorize(ctx, PolicyReq{
-				Subject:     userID,
-				SubjectType: UserType,
-				SubjectKind: UsersKind,
-				Permission:  AdminPermission,
-				Object:      id,
-				ObjectType:  DomainType,
-			}); err != nil {
-				// Append all non-admins to ids.
-				ids = append(ids, userID)
+			pr.Subject = userID
+			if err := svc.Authorize(ctx, pr); err == nil {
+				// Non admin attempts to remove admin.
+				return errors.Wrap(svcerr.ErrAuthorization, err)
 			}
 		}
-
-		userIds = ids
 	}
 
-	for _, rel := range []string{MemberRelation, ViewerRelation, EditorRelation} {
-		// Remove only non-admins.
-		if err := svc.removeDomainPolicies(ctx, id, rel, userIds...); err != nil {
-			return err
+	for _, userID := range userIds {
+		if err := svc.DeletePolicy(ctx, PolicyReq{
+			Subject:     EncodeDomainUserID(id, userID),
+			SubjectType: UserType,
+			SubjectKind: UsersKind,
+			Object:      id,
+			ObjectType:  DomainType,
+		}); err != nil {
+			return errors.Wrap(errRemovePolicies, err)
```

Comment:
@WashingtonKK 
They are not removed from Repo . I could not see repo to call for removing users from Database.